lib/converters/v1.0.0/converter-v1-to-v2.js

```diff
@@ -272,9 +272,12 @@ _.assign(Builders.prototype, {
      */
     auth: function (entityV1, options) {
         if (!entityV1) { return; }
-        if (util.notLegacy(entityV1, 'auth')) { return util.authArrayToMap(entityV1, options); }
-        if ((entityV1.currentHelper === null) || (entityV1.currentHelper === 'normal')) { return null; }
-        if (!entityV1.currentHelper) { return; }
+        if ((util.notLegacy(entityV1, 'auth') || this.options.prioritizeV2) && entityV1.auth) {
```

Comment:
@kunagpal could you flip this checks to 

```
if (prioritizeV2) {
  if (entityV1.auth) {}
  else if (entityV1.currentHelper) {}// or a more accurate legacy check
  else
}
else {
  if (entityV1.currentHelper) {}// or a more accurate legacy check
  else if (entityV1.auth) {}
  else
}
```

Reply:
That makes the code a lot more verbose 😓

Reply:
Yeah but it makes it easy to reason about, and reduces possible bugs.